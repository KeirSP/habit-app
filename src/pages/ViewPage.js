import React, { Component } from 'react';
import HabiList from '../components/HabitList'
import Form from '../components/Form'

class ViewPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            // userID: this.props.location.state.userID
        }
    }
    
    render() {
        return (
            <div>
                <HabiList props={{userID: 1}}/>
                <Form/>
            </div>
        );
    }
}

export default ViewPage;