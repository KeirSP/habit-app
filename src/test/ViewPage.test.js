import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import ViewPage from '../pages/ViewPage'

describe('<ViewPage/>', () => {

  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<ViewPage />, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  let wrapper;
  beforeEach(() => wrapper = shallow(<ViewPage />));
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(1);
  });
  it('should render 4 Routes to App Component', () =>{
    expect(wrapper.find('Route').length).toEqual(2);
})
})