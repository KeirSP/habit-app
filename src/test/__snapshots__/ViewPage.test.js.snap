// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewPage/> Renders correctly 1`] = `undefined`;
